README.md

```diff
@@ -139,6 +142,89 @@ To upgrade this resource from `4.0.6` to `4.2.7`:
+     <metadata.name of the MongoDB resource>-2.<metadata.name of the MongoDB resource>-svc.<namespace>.svc.cluster.local
+     ```
+
+1. Create a Kubernetes ConfigMap that contains the certificate for the CA that signed your server certificate. For a certificate file named `ca.crt`:
+   ```
+   kubectl create configmap <tls-ca-configmap-name> --from-file=ca.crt --namespace <namespace>
+   ```
+
+1. Create a Kubernetes secret that contains the server certificate and key for the members of your replica set. For a server certificate named `server.crt` and key named `server.key`:
+   ```
+   kubectl create secret tls <tls-secret-name> --cert=server.crt --key=server.key --namespace <namespace>
+   ```
+
+### Procedure
+
+To secure connections to MongoDB resources using TLS:
+
+1. Add the following fields to the MongoDB resource definition:
+
+   - `security.tls.enabled`: Encrypts communications using TLS certificates between MongoDB hosts in a replica set and client applications and MongoDB deployments. Set to `true`.
+   - `security.tls.optional`: (**Optional**) Enables the members of the replica set to accept both TLS and non-TLS client connections. Equivalent to setting the MongoDB[`net.tls.mode`](https://docs.mongodb.com/manual/reference/configuration-options/#net.tls.mode) setting to `preferSSL`. If omitted, defaults to `false`.
+
+     ---
+     **NOTE**
+
+     When you enable TLS on an existing replica set deployment:
+
+     a. Set `security.tls.optional` to `true`.
+     b. Apply the configuration to Kubernetes.
+     c. Upgrade your existing clients to use TLS.
+     d. Set `security.tls.optional` to `false`.
+     e. Complete the remaining steps in the procedure.
+
+     ---
+   - `security.tls.certificateKeySecretRef.name`: Name of the Kubernetes secret that contains the server certificate and key that you created in the [prerequisites](#prerequisites-1).
+   - `security.tls.caConfigMapRef.name`: Name of the Kubernetes ConfigMap that contains the Certificate Authority certificate used to sign the server certificate that you created in the [prerequisites](#prerequisites-1).
+
+   ```yaml
+   security:
+     tls:
+       enabled: true
+       optional: false
+       certificateKeySecretRef:
+         name: <tls-secret-name>
+       caConfigMapRef:
+         name: <tls-ca-configmap-name>
+   ```
+
+1. Apply the configuration to Kubernetes:
+   ```
+   kubectl apply -f <example>.yaml --namespace <my-namespace>
+   ```
+1. From within the Kubernetes cluster, connect to the MongoDB resource.
+   - If `security.tls.optional` is omitted or `false`: clients must
+     establish TLS connections to the MongoDB servers in the replica set.
+   - If `security.tls.optional` is true, clients can establish TLS or
+     non-TLS connections to the MongoDB servers in the replica set.
+
+   See the documentation for your connection method to learn how to establish a TLS connection to a MongoDB server.
+
 ## Supported Features
```

Comment:
Q: Do we need to add anything here about TLS support between MDB resources? Might not be necessary

Reply:
amended line below to be more explicit about this

Reply:
I don't think so. Once TLS is enabled, server-server encryption should just work and the user doesn't have to worry about it.

Reply:
Didn’t see your comment when writing mine but that’s a nice way to clarify!

Reply:
see https://github.com/mongodb/mongodb-kubernetes-operator/pull/128/files#diff-04c6e90faac2675aa89e2176d2eec7d8R243. I'm happy to revert this change if you like.

Reply:
I like it, no need to change!